### DISCUSSION TITLE
preload symfony/cache file will cause php-fpm exit with code 70

### DISCUSSION THREAD

**Discussion by everyx** [symfony/symfony#47382] [Q&A]
### Symfony version(s) affected

6.0.11

### Description

When upgrade Symfony cache from 5.x to 6.0.x, the php-fpm start failed with code `70` with opcache preload file enable.

### How to reproduce

Use this preload file bellow with php v8.0.22

```php
<?php

$files = [
    // ***
    '/var/www/vendor/psr/cache/src/CacheItemInterface.php',
    '/var/www/vendor/psr/cache/src/CacheItemPoolInterface.php',
    '/var/www/vendor/symfony/cache-contracts/ItemInterface.php',
    '/var/www/vendor/symfony/cache/CacheItem.php',
    '/var/www/vendor/symfony/cache/Adapter/AdapterInterface.php',
    '/var/www/vendor/symfony/cache/Adapter/TagAwareAdapterInterface.php',
    // ***
];

foreach ($files as $file) {
    try {
        if (!(is_file($file) && is_readable($file))) {
            continue;
        }
        opcache_compile_file($file);
    } catch (\Throwable $e) {
        echo 'Preloader Script has stopped with an error:' . \PHP_EOL;
        echo 'Message: ' . $e->getMessage() . \PHP_EOL;
        echo 'File: ' . $file . \PHP_EOL;

        throw $e;
    }
}
```

### Additional Context

1. php version: 8.0.22
2. os: Alpine linux 3.16

**Comment by MatTheCat** [comment]
This may be a permission issue. Which user runs FPM? What is your `opcache.preload_user` setting?

**Reply by everyx** [reply]
> This may be a permission issue. Which user runs FPM? What is your `opcache.preload_user` setting?

It's no permission issue, I use php:8.0.22-fpm-alpine3.16 docker image to run my application, there is no issues before, until I upgrade project dependencies symfony packages from 5.x to 6.x, by the way, I use https://github.com/diego-ninja/Preloader to auto generate preload file.

**Reply by MatTheCat** [reply]
Okay, I somewhat reproduced (I got code 139 instead of 70 though) and the bug seems to appear before PHP 8.1 because of the `CacheItem` class being present in your preloading script as well as in https://github.com/symfony/symfony/blob/2bda6243b1e4da3d60063de1f47a3eeecd4cdad6/src/Symfony/Component/Cache/Adapter/AdapterInterface.php#L17-L18

If it is true, you’ll have to fill an issue on https://github.com/diego-ninja/Preloader and/or upgrade your PHP version.

@nicolas-grekas may I ask you if it rings any bell?

**Reply by nicolas-grekas** [reply]
Maybe related to the preloading issue that made us [adopt PHP 8.1](https://github.com/symfony/symfony/pull/45377) recently?

**Reply by everyx** [reply]
@MatTheCat yes i upgraded the PHP version to 8.1.x, but this brings me to  [another bug of PHP](https://github.com/php/php-src/issues/7817) 😂, thank you for your reply.

**Reply by MatTheCat** [reply]
@nicolas-grekas thanks, doesn't seem related though 😞

@everyx okay, looks like you can disable JIT as a workaround.
